From b38d1e9422ffef61c63571581b55cc794a5e6a39 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Wed, 25 Mar 2015 18:14:40 +0800
Subject: [PATCH 0180/5242] MLK-10461: arm: imx7: set the PFD override bit
 before system entering low power mode

commit  bd31d7d1513934242c9c364e4a2c91cc81666613 from
https://source.codeaurora.org/external/imx/linux-imx.git

the PFD override bit must be set before system entering any low power mode.

Signed-off-by: Bai Ping <b51503@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/anatop.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/anatop.c b/arch/arm/mach-imx/anatop.c
index 3459858..36430ce 100644
--- a/arch/arm/mach-imx/anatop.c
+++ b/arch/arm/mach-imx/anatop.c
@@ -83,10 +83,10 @@ static inline void imx_anatop_disconnect_high_snvs(bool enable)
 void imx_anatop_pre_suspend(void)
 {
 	if (cpu_is_imx7d()) {
-		/* PLL overwrite set */
+		/* PLL and PFDs overwrite set */
 		regmap_write(anatop, ANADIG_ARM_PLL + REG_SET, 1 << 20);
 		regmap_write(anatop, ANADIG_DDR_PLL + REG_SET, 1 << 19);
-		regmap_write(anatop, ANADIG_SYS_PLL + REG_SET, 1 << 17);
+		regmap_write(anatop, ANADIG_SYS_PLL + REG_SET, 0x1ff << 17);
 		regmap_write(anatop, ANADIG_ENET_PLL + REG_SET, 1 << 13);
 		regmap_write(anatop, ANADIG_AUDIO_PLL + REG_SET, 1 << 24);
 		regmap_write(anatop, ANADIG_VIDEO_PLL + REG_SET, 1 << 24);
@@ -107,10 +107,10 @@ void imx_anatop_pre_suspend(void)
 void imx_anatop_post_resume(void)
 {
 	if (cpu_is_imx7d()) {
-		/* PLL overwrite set */
+		/* PLL and PFDs overwrite clear */
 		regmap_write(anatop, ANADIG_ARM_PLL + REG_CLR, 1 << 20);
 		regmap_write(anatop, ANADIG_DDR_PLL + REG_CLR, 1 << 19);
-		regmap_write(anatop, ANADIG_SYS_PLL + REG_CLR, 1 << 17);
+		regmap_write(anatop, ANADIG_SYS_PLL + REG_CLR, 0x1ff << 17);
 		regmap_write(anatop, ANADIG_ENET_PLL + REG_CLR, 1 << 13);
 		regmap_write(anatop, ANADIG_AUDIO_PLL + REG_CLR, 1 << 24);
 		regmap_write(anatop, ANADIG_VIDEO_PLL + REG_CLR, 1 << 24);
-- 
1.7.9.5

